modules/stanford_earth_capx/src/Plugin/migrate/process/StanfordEarthCapProfileImage.php

```diff
@@ -0,0 +1,132 @@
+ * Additionally, a special value is available to represent the filename of
+ * the file '!file'. Useful to just populate the alt or title field with the
+ * filename.
+ *
+ * @see Drupal\migrate\Plugin\migrate\process\Get
+ *
+ * @see Drupal\migrate_file\Plugin\migrate\process\FileImport.php
+ *
+ * Example:
+ *
+ * @code
+ * destination:
+ *   plugin: entity:node
+ * source:
+ *   # assuming we're using a source plugin that lets us define fields like this
+ *   item_selector: values
+ *   fields:
+ *     -
+ *       name: image
+ *       label: 'Main Image'
+ *       selector: profilePhotos/bigger/url
+ *   constants:
+ *     file_destination: 'public://path/to/save/'
+ * process:
+ *   title:
+ *     plugin: default_value
+ *     default_value: Some Title
+ *   uid:
+ *     plugin: default_value
+ *     default_value: 1
+ *   field_image:
+ *     plugin: stanford_earth_cap_profile_image
+ *     source: image
+ *     destination: constants/file_destination
+ *     uid: @uid
+ *     title: title
+ *     alt: !file
+ *     skip_on_missing_source: true
+ *
+ *
+ * @endcode
+ *
+ * @see \Drupal\migrate\Plugin\MigrateProcessInterface
+ *
+ * @MigrateProcessPlugin(
+ *   id = "stanford_earth_cap_profile_image"
+ * )
+ */
+class StanfordEarthCapProfileImage extends FileImport {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, array $plugin_definition, StreamWrapperManagerInterface $stream_wrappers, FileSystemInterface $file_system, MigrateProcessInterface $download_plugin) {
+    $configuration += [
+      'title' => NULL,
+      'alt' => NULL,
+      'width' => NULL,
+      'height' => NULL,
+      'destination_property_entity' => NULL,
+      'destination_property_bundle' => NULL,
+    ];
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $stream_wrappers, $file_system, $download_plugin);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+
+    // see if we already have the current profile photo
+    $info = new EarthCapxInfo($row->getSourceProperty('sunetid'));
+    $photoId = $info->currentProfilePhotoId($row->getSource());
+    if ($photoId) {
+      $value = ['target_id' => $photoId];
+    } else {
+      $this->configuration['id_only'] = FALSE;
+      $value = parent::transform($value, $migrate_executable, $row, $destination_property);
+    }
+
+    if ($value && is_array($value)) {
+      // Add the image field specific sub fields.
+      foreach (['title', 'alt', 'width', 'height'] as $key) {
+        if ($property = $this->configuration[$key]) {
+          if ($property == '!file') {
+            $file = File::load($value['target_id']);
+            $value[$key] = $file->getFilename();
+          }
+          else {
+            $value[$key] = $this->getPropertyValue($property, $row);
+          }
+        }
+      }
+      return $value;
+    }
+    else {
```

Comment:
This else statement is not needed.

Reply:
This was copied from the plugin from the migrate_file module, but I've removed it here.

Reply:
They might like a patch :)